File: src/main/java/kr/co/bookand/backend/issue/controller/IssueController.java
package kr.co.bookand.backend.issue.controller;

import io.swagger.annotations.Api;
import io.swagger.annotations.ApiOperation;
import kr.co.bookand.backend.common.domain.Message;
import kr.co.bookand.backend.common.domain.dto.PageResponse;
import kr.co.bookand.backend.issue.service.IssueService;
import lombok.RequiredArgsConstructor;
import org.springframework.data.domain.Pageable;
import org.springframework.data.web.PageableDefault;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import static kr.co.bookand.backend.issue.domain.dto.IssueDto.*;

@RestController
@RequestMapping("/api/v1/issues")
@RequiredArgsConstructor
@Api(tags = "오류 및 버그 API")
public class IssueController {
    private final IssueService issueService;

    @ApiOperation(value = "오류 및 버그 신고 생성")
    @PostMapping("")
    public ResponseEntity<IssueIdResponse> createIssue(
            @ModelAttribute CreateIssueRequest createIssueRequest
    ) {
        return ResponseEntity.ok(issueService.createIssue(createIssueRequest));
    }

    @ApiOperation(value = "오류 및 버그 신고 조회 (WEB)")
    @GetMapping("")
    public ResponseEntity<PageResponse<IssueSimpleResponse>> getIssueList(
            @PageableDefault Pageable pageable
    ) {
        return ResponseEntity.ok(issueService.getIssueList(pageable));
    }

    @ApiOperation(value = "오류 및 버그 신고 상세 조회")
    @GetMapping("/{issueId}")
    public ResponseEntity<IssueResponse> getIssue(
            @PathVariable Long issueId
    ) {
        return ResponseEntity.ok(issueService.getIssue(issueId));
    }

    @ApiOperation(value = "오류 및 버그 신고 체크 하기")
    @PutMapping("/{issueId}/check")
    public ResponseEntity<Message> checkIssue(
            @PathVariable Long issueId,
            @RequestParam Boolean checkConfirmed
    ) {
        return ResponseEntity.ok(issueService.checkConfirmed(issueId, checkConfirmed));
    }

}

File: src/main/kotlin/kr/co/bookand/backend/issue/controller/IssueController.kt
package kr.co.bookand.backend.issue.controller

import io.swagger.annotations.Api
import io.swagger.annotations.ApiOperation
import kr.co.bookand.backend.account.service.AccountService
import kr.co.bookand.backend.common.model.MessageResponse
import kr.co.bookand.backend.issue.dto.CreateIssueRequest
import kr.co.bookand.backend.issue.dto.IssueIdResponse
import kr.co.bookand.backend.issue.dto.IssueResponse
import kr.co.bookand.backend.issue.dto.IssueSimpleListResponse
import kr.co.bookand.backend.issue.service.IssueService
import org.springframework.data.domain.Pageable
import org.springframework.data.web.PageableDefault
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/api/v1/issues")
@Api(tags = ["오류 및 버그 API"])
class IssueController(
    private val issueService: IssueService,
    private val accountService: AccountService
) {
    @ApiOperation(value = "오류 및 버그 신고 생성")
    @PostMapping("")
    fun createIssue(
        @ModelAttribute createIssueRequest: CreateIssueRequest
    ): ResponseEntity<IssueIdResponse> {
        val account = accountService.getCurrentAccount()
        return ResponseEntity.ok(issueService.createIssue(account, createIssueRequest))
    }

    @ApiOperation(value = "오류 및 버그 신고 조회 (WEB)")
    @GetMapping("")
    fun getIssueList(
        @PageableDefault pageable: Pageable
    ): ResponseEntity<IssueSimpleListResponse> {
        val account = accountService.getCurrentAccount()
        return ResponseEntity.ok(issueService.getIssueList(account, pageable))
    }

    @ApiOperation(value = "오류 및 버그 신고 상세 조회")
    @GetMapping("/{issueId}")
    fun getIssue(
        @PathVariable issueId: Long
    ): ResponseEntity<IssueResponse> {
        return ResponseEntity.ok(issueService.getIssue(issueId))
    }

    @ApiOperation(value = "오류 및 버그 신고 체크 하기")
    @PutMapping("/{issueId}/check")
    fun checkIssue(
        @PathVariable issueId: Long,
        @RequestParam checkConfirmed: Boolean
    ): ResponseEntity<IssueIdResponse> {
        return ResponseEntity.ok(issueService.checkConfirmed(issueId, checkConfirmed))
    }
}